File: android/src/main/kotlin/live/hms/hmssdk_flutter/HMSSimulcastLayerSettingsPolicyExtension.kt
package live.hms.hmssdk_flutter

import live.hms.video.sdk.models.role.LayerParams

class HMSSimulcastLayerSettingsPolicyExtension {

    companion object {
        fun toDictionary(policy: LayerParams): HashMap<String, Any?> {
            val hashMap = HashMap<String, Any?>()

            if (policy.rid == null) {
                return hashMap
            }
            hashMap["rid"] = policy.rid
            hashMap["max_bitrate"] = policy.maxBitrate ?: 0
            hashMap["max_frame_rate"] = policy.maxFramerate ?: 0
            hashMap["scale_resolution_down_by"] = policy.scaleResolutionDownBy ?: 0
            return hashMap
        }
    }
}

File: packages/hmssdk_flutter/ios/Classes/Models/HMSSimulcastLayerSettingsPolicyExtension.swift
//
//  HMSSimulCastSettingsExtension.swift
//  hmssdk_flutter
//
//  Copyright © 2021 100ms. All rights reserved.
//

import Foundation
import HMSSDK

class  HMSSimulcastLayerSettingsPolicyExtension {

    static func toDictionary(_ policy: HMSSimulcastLayerSettingsPolicy) -> [String: Any] {
        [
            "rid": policy.rid,
            "max_bitrate": policy.maxBitrate ?? 0,
            "max_frame_rate": policy.maxFramerate ?? 0,
            "scale_resolution_down_by": policy.scaleResolutionDownBy ?? 0
        ]
    }
}
